youtube_dl/extractor/pandoratv.py

```diff
@@ -0,0 +1,37 @@
+# encoding: utf-8
+from __future__ import unicode_literals
+
+import re
+import json
+
+from .common import InfoExtractor
+
+class PandoraTVIE(InfoExtractor):
+    _VALID_URL = r'http://(.*?\.)?channel.pandora.tv/channel/video.ptv\?.*?prgid=(?P<prgid>.*?)&.*'
```

Comment:
`ch_userid` is mandatory, without it 404 is returned, e.g. http://channel.pandora.tv/channel/video.ptv?prgid=53294230&ref=main&lot=cate_01_2 shouldn't be matched.
`http://(.*?\.)?` should be `http://(?:.+?\.)?`.
`(?P<prgid>.*?)` should be `(?P<prgid>.+?)`.
`.` representing dots should be `\.`.